ecaudit/src/test/java/com/ericsson/bss/cassandra/ecaudit/auth/TestWrappingAuditAuthenticator.java

```diff
@@ -0,0 +1,174 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.auth;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ericsson.bss.cassandra.ecaudit.AuditAdapter;
+import com.ericsson.bss.cassandra.ecaudit.common.record.Status;
+import com.ericsson.bss.cassandra.ecaudit.config.AuditConfig;
+import com.ericsson.bss.cassandra.ecaudit.test.mode.ClientInitializer;
+import org.apache.cassandra.auth.AuthenticatedUser;
+import org.apache.cassandra.auth.IAuthenticator;
+import org.apache.cassandra.exceptions.AuthenticationException;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class TestWrappingAuditAuthenticator
+{
+    @Mock
+    IAuditAuthenticator mockAuthenticator;
+
+    @Mock
+    AuditAdapter mockAuditAdapter;
+
+    @Mock
+    AuditConfig mockConfig;
+
+    @Mock
+    IAuditAuthenticator.AuditSaslNegotiator mockSaslNegotiator;
+
+    @InjectMocks
+    WrappingAuditAuthenticator authenticator;
+
+    @BeforeClass
+    public static void beforeClass()
+    {
+        ClientInitializer.beforeClass();
+    }
+
+    @Before
+    public void setup()
+    {
+        when(mockAuthenticator.newAuditSaslNegotiator()).thenReturn(mockSaslNegotiator);
+    }
+
+    @After
+    public void after()
+    {
+        verifyNoMoreInteractions(mockAuthenticator, mockAuditAdapter);
+    }
+
+    @AfterClass
+    public static void afterClass()
+    {
+        ClientInitializer.afterClass();
+    }
+
+    @Test
+    public void testWrappedMethodsAreDelegated()
+    {
+        authenticator.setup();
+        authenticator.protectedResources();
+        authenticator.requireAuthentication();
+        authenticator.validateConfiguration();
+        authenticator.legacyAuthenticate(Collections.emptyMap());
+        authenticator.alterableOptions();
+        authenticator.supportedOptions();
```

Comment:
it is verified that the method calls are delegated... But not that the returned values are passed back from the wrapped authenticator.

Maybe you could setup the wrapped authenticator with some mocked data to assert against?